.github/workflows/ci.yml

```diff
@@ -0,0 +1,72 @@
+name: CI
+
+on:
+  push:
+    branches: [master]
+    tags: ['*']
+  pull_request:
+    branches: [master]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    environment: ci
+    strategy:
+      matrix:
+        ruby-version: ['2.7', '3.0.6', '3.1', '3.2']
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby-version }}
+        bundler-cache: true
+        bundler: '2.4.22'
```

Comment:
This bundler line is not necessary, as setup-ruby will derive the version needed automatically based on the ruby version.